### ISSUE TITLE
Skip the first raid after a reset

### ISSUE THREAD

**Issue by CardBreaker** [LegoFigure11/RaidCrawler#169]
The game generates raids for both the current day and the next day. This means that every reset, you're wasting time by scanning the same first raids. My suggested improvement is that after every reset, advance 1 day and save.

**Comment by CardBreaker** [comment]
Documenting some extra considerations here:

- You could try to optimize this by simply saving before the reset to avoid an extra skip. However, the game gets laggier the more skips someone wants to reset after. I don't think I'd trust saving to work in that case, so it's not a reliable solution.
- You could simply save after the first skip both after starting to advance as well as after resetting the game to avoid the extra skip. This might actually be a better way to go.

**Comment by CardBreaker** [comment]
Actually, scratch that second bullet point. That would still result in a raid being redundantly scanned every reset.

**Comment by CardBreaker** [comment]
After more thought, saving needs to work under intense lag. Let's say the user has the save on match setting turned on and has the number of skips to reset after set to 100. If they find a match on skip 99, we need to make sure that save goes through in case the user is away and the date rolls over. With that in mind, it's most efficient if we just save before resetting which should be faster than advancing and saving after a reset even if we have to extend some of the save delays.

I did some testing on my v1 Switch by making this change and increasing the skips per reset by increments of 10. I started having saving issues at 70 skips per reset. Maybe we could simply add a configurable delay to the save menuing that the user could set based on their skips per reset setting as well as their Switch's performance.
